src/m365/spo/commands/tenant/tenant-site-membership-list.spec.ts

```diff
@@ -0,0 +1,385 @@
+    },
+    {
+      email: 'visitor2Email@email.com',
+      loginName: 'i:0#.f|membership|visitor2loginName@email.com',
+      name: 'visitor2DisplayName',
+      userPrincipalName: 'visitor2loginName',
+      associatedGroupType: 'Visitor'
+    }
+  ];
+  const adminUrl = 'https://contoso-admin.sharepoint.com';
+  const siteUrl = 'https://contoso.sharepoint.com/sites/site';
+  const siteId = '00000000-0000-0000-0000-000000000010';
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    sinon.stub(spo, 'getSpoAdminUrl').resolves(adminUrl);
+    auth.connection.active = true;
+    auth.connection.spoUrl = 'https://contoso.sharepoint.com';
+    commandInfo = cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      request.post,
+      spo.getSiteAdminPropertiesByUrl
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.connection.active = false;
+    auth.connection.spoUrl = undefined;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.TENANT_SITE_MEMBERSHIP_LIST);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('passes validation if the role option is a valid role', async () => {
+    const actual = await command.validate({ options: { siteUrl: 'https://contoso.sharepoint.com', role: 'Owner' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation if the siteUrl option is not a valid SharePoint URL', async () => {
+    const actual = await command.validate({ options: { siteUrl: 'foo' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the role option is not a valid role', async () => {
+    const actual = await command.validate({ options: { siteUrl: 'https://contoso.sharepoint.com', role: 'foo' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('lists all site membership groups', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${adminUrl}/_api/SPO.Tenant/sites/GetSiteUserGroups?siteId='${siteId}'&userGroupIds=[0,1,2]`) {
+        return { value: [{ userGroup: ownerMembershipList }, { userGroup: membersMembershipList }, { userGroup: visitorsMembershipList }] };
+      };
+
+      throw 'Invalid request';
+    });
+
+    sinon.stub(spo, 'getSiteAdminPropertiesByUrl').resolves({ SiteId: siteId } as any);
```

Comment:
Since we have to use this in almost every test, let's move it to the `before` hook so we avoid repetition.